legacy/application/models/ShowInstance.php

```diff
@@ -844,4 +844,30 @@ public function isRepeating()
     {
         return $this->getShow()->isRepeating();
     }
+
+    public function trimOverbooked()
+    {
+        // iterate through all of the showinstance items and delete those with start time greater than endtime of the show
+        $showEnd = new DateTime($this->getShowInstanceEnd(), new DateTimeZone('UTC'));
+        $removeItems = [];
+        foreach ($this->getShowListContent('UTC') as $item) {
+            $itemStart = new DateTime($item['starts'], new DateTimeZone('UTC'));
+            if ($itemStart->getTimestamp() > $showEnd->getTimestamp()) {
+                $removeItems[] = $item['id'];
+            }
+        }
+
+        // now remove the items
+        $sql = <<<'SQL'
+        DELETE FROM cc_schedule
+         WHERE id in (:ids)
+           AND instance_id = :instance_id
+           AND playout_status >= 0
+        SQL;
+
+        return Application_Common_Database::prepareAndExecute($sql, [
+            ':ids' => implode($removeItems, ','),
```

Comment:
For your information, this should be `implode(',', $removeItems)`.

I've already fixed it.